### PR TITLE
community rewards

### PR DESCRIPTION
**Pull request by mikluke**

### CODE REVIEW

File: x/token/keeper/querier.go [hunk 1]
```diff
@@ -37,14 +45,29 @@ func queryBalance(ctx sdk.Context, path []string, _ abci.RequestQuery, keeper Ke
...
+}
+
+// nolint: unparam
+func queryPool(ctx sdk.Context, _ []string, _ abci.RequestQuery, keeper Keeper) ([]byte, error) {
+	interval := keeper.GetParams(ctx).RewardsBlockInterval
 
-	res, err := codec.MarshalJSONIndent(keeper.cdc, Balance{
-		Balance: balance,
+	bz, err := keeper.cdc.MarshalBinaryBare(Pool{
+		Size:                       keeper.distributionKeeper.GetFeePoolCommunityCoins(ctx),
```
Comment (megaherz) [inline x/token/keeper/querier.go:64]: What is the decentrd command to update FeePoolCommunityCoins? It is populated as transaction fee is applied as well?

File: x/token/keeper/querier.go [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
 type Balance struct {
-	Balance float64 `json:"balance" amino:"unsafe"`
+	Balance      sdk.Dec `json:"balance"`
+	BalanceDelta sdk.Dec `json:"balanceDelta"`
+}
+
+type Pool struct {
+	Size                       sdk.DecCoins `json:"size"`
+	TotalDelta                 sdk.Dec      `json:"totalDelta"`
+	ExpectedDistributionHeight int64        `json:"expectedDistributionHeight"`
```
Comment (megaherz) [inline x/token/keeper/querier.go:24]: Maybe NextDistributionHeight?

File: x/token/keeper/querier.go [hunk 3]
```diff
@@ -1,21 +1,27 @@
...
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-
-	"github.com/Decentr-net/decentr/x/utils"
 )
 
 const (
 	QueryBalance = "balance"
+	QueryPool    = "pool"
```
Comment (megaherz) [inline x/token/keeper/querier.go:13]: Is there a way to query PDV reward distribution history of the specific account?

File: x/token/module.go [hunk 4]
```diff
@@ -98,7 +98,11 @@ func (am AppModule) NewQuerierHandler() sdk.Querier {
 	return NewQuerier(am.keeper)
 }
 
-func (am AppModule) BeginBlock(_ sdk.Context, _ abci.RequestBeginBlock) {}
+func (am AppModule) BeginBlock(ctx sdk.Context, b abci.RequestBeginBlock) {
+	if b.Header.Height%am.keeper.GetParams(ctx).RewardsBlockInterval == 0 {
```
[the remark this answers is not in the document]
Comment (mikluke) [reply x/token/module.go:102]: No, it looks like it won't break anything even it happens

File: x/token/keeper/querier.go (hunk 1 above)
Comment (mikluke) [reply x/token/keeper/querier.go:64]: FundCommunityPool as I remember.

Yes, there is a parameter of distribution module 'community_tax' that means how make stakes should be taken

File: x/token/keeper/querier.go (hunk 3 above)
Comment (mikluke) [reply x/token/keeper/querier.go:13]: Oops

**Pull request merged**